### PR TITLE
Enhancement: better monetary field with currency code

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Lets [Angular](https://angular.io/api/common/CurrencyPipe) handle the formatting including default currency from the current locale. Should be backwards compatible because it still accepts the number format but new format is e.g. `GBP124.11`

Screenshot 2024-02-22 at 11 06 58 AM
Screenshot 2024-02-22 at 11 06 38 AM
Screenshot 2024-02-22 at 11 06 08 AM

And if e.g. youre in Europe:
Screenshot 2024-02-22 at 11 50 25

Closes #5855

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -406,7 +406,7 @@ The following custom field types are supported:
 - `URL`: a valid url
 - `Integer`: integer number e.g. 12
 - `Number`: float number e.g. 12.3456
-- `Monetary`: float number with exactly two decimals, e.g. 12.30
+- `Monetary`: [ISO 4217 country code](https://en.wikipedia.org/wiki/ISO_4217#List_of_ISO_4217_currency_codes) and a number with exactly two decimals, e.g. USD12.30
```
Comment (marcelbrueckner) [inline docs/usage.md:409]: Link text says "country" instead of "currency"

Comment (shamoon) [reply docs/usage.md:409]: Thanks =)

File: src/documents/migrations/1045_alter_customfieldinstance_value_monetary.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("documents", "1044_workflow_workflowaction_workflowtrigger_and_more"),
+    ]
+
+    operations = [
+        migrations.AlterField(
```
Comment (stumpylog) [inline src/documents/migrations/1045_alter_customfieldinstance_value_monetary.py:14]: Will this correctly change existing decimals to strings?

Comment (shamoon) [reply src/documents/migrations/1045_alter_customfieldinstance_value_monetary.py:14]: Yea it worked fine testing with SQLite but youre right I should make a point to test with e.g. Postgres, there were some quirks with that recently.

before:
Screenshot 2024-02-26 at 4 52 15 PM
Screenshot 2024-02-26 at 4 52 21 PM

after:
Screenshot 2024-02-26 at 4 53 37 PM
Screenshot 2024-02-26 at 4 53 47 PM

Comment (shamoon) [reply src/documents/migrations/1045_alter_customfieldinstance_value_monetary.py:14]: Ok yea, tested with Postgres and was OK too. Let me know if you notice anything tho!

Screenshot 2024-02-26 at 5 04 13 PM
Screenshot 2024-02-26 at 5 05 45 PM

**Review by stumpylog — APPROVED** [review]

**Pull request merged**